Fix set artwork. Remove warning for RN 0.50. Fix set listener

kesha-antonov opened the pull request:
Hello! @tanguyantoine 

#### What's this PR do?

1) It cleans up code and set correct artwork when ```updatePlayback```/```setPlayback``` called
2) Remove warning for RN 0.50
3) Fix set listener of "RNMusicControlEvent". We should set it only if it not existed. Not on each call of "on"

tanguyantoine commented on ios/MusicControlManager.m:
```diff
@@ -231,36 +236,41 @@ - (void)sendEvent:(NSString*)event {
+  if (artwork) {
+      if ([artwork isKindOfClass:[NSString class]]) {
+           artworkUrl = artwork;
+      } else if ([[artwork valueForKey: @"uri"] isKindOfClass:[NSString class]]) {
```
If you expect artwork to be a `NSString` type (from parameter type) then we won't ever pass in this condition. Right ?

kesha-antonov replied:
I'm not obj-c expert. But it works

tanguyantoine commented:
Thank you for the PR. May I ask you to split it in multiple PR?

kesha-antonov commented:
Hello!
Any trouble understanding code?

tanguyantoine commented:
nope, but now there is conflict 😁

tanguyantoine commented:
@DanNi0130 may I ask you to review?

DanNi0130 commented:
I'll take a look at this tomorrow morning!

kesha-antonov commented:
Resolved

tanguyantoine commented on index.ios.js:
```diff
@@ -30,22 +38,20 @@ var MusicControl = {
   setPlayback: function (info) {
-    // Backwards compatibility. Use updatePlayback instead.
+    info = formatInfo(info);
     NativeMusicControl.updatePlayback(info);
```
Could you remove variable creation like this

```js
NativeMusicControl.updatePlayback(formatInfo(info));
```

tanguyantoine commented on index.ios.js:
```diff
@@ -30,22 +38,20 @@ var MusicControl = {
   },
   updatePlayback: function(info) {
+    info = formatInfo(info);
     NativeMusicControl.updatePlayback(info);
```
same

tanguyantoine commented on index.ios.js:
```diff
@@ -30,22 +38,20 @@ var MusicControl = {
-    }
-
+    info = formatInfo(info);
     NativeMusicControl.setNowPlaying(info);
```
same

kesha-antonov commented:
@tanguyantoine I've updated what you asked.
Also I've noticed that there's ```index.js.android``` file which is almost the same. So I refactored ```index.js.ios/android``` into one ```index.js``` to prevent copy-paste code

tanguyantoine commented:
Please see #79 for file merging. @DanNi0130 if it is okay for you, feel free to merge (without index merging, another PR is preferable)

kesha-antonov commented:
Oops. Missed it. Alright guys. It's up to you to decide what to do next

kesha-antonov commented:
I think it's better to merge my PR. Or you'll have to resolve conflicts 2 times

DanNi0130 commented:
@tanguyantoine, this looks fine to me. Should I merge this?

tanguyantoine commented:
I would rather not merging this PR because of the index merge. If it was done in 2 PR then yes (easier to rollback if necessary)

Again, the fix is :+1: and thank to you. But this PR does well too much. 

If @Guichaguri agreed then let’s go.

Guichaguri commented:
The index looks a bit messier, I agree that it should be moved to another PR, but I don't see any problems with this being merged

kesha-antonov commented:
Come on guys. It's not a rocket science. Not much code changed

tanguyantoine commented:
Rocket Science or not We prefer to have two different PRs. We sincerely appreciate your stuff but 2 PR make more sense

kesha-antonov commented:
Ok. Sorry I do not have time now for this

tanguyantoine commented:
I will try to backport you changes is I find the time.

kesha-antonov commented:
Ok thanks!
I'll try to separate these to PR when I get back to this lib

kesha-antonov commented:
I see this not getting merged. Closing
